File: 11.盛最多水的容器.go
/*
 * @lc app=leetcode.cn id=11 lang=golang
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
func maxArea(height []int) int {
	len := len(height)
	ans := 0
	for left, right := 0, len-1; left < right; {
		leftNum := height[left]
		rightNum := height[right]
		length := right - left
		var minHeight int
		if leftNum > rightNum {
			minHeight = rightNum
			right--
		} else {
			minHeight = leftNum
			left++
		}
		area := length * minHeight
		if area > ans {
			ans = area
		}
	}
	return ans
}

// @lc code=end

File: 11.盛最多水的容器.py
from typing import *
#
# @lc app=leetcode.cn id=11 lang=python3
#
# [11] 盛最多水的容器
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        ans = 0
        while left < right:
            left_num = height[left]
            right_num = height[right]
            length = right - left
            if left_num > right_num:
                min_num = right_num
                right -= 1
            else:
                min_num = left_num
                left += 1
            area = min_num * length
            ans = max(ans ,area)
        return ans
# @lc code=end
